boards/BMS/Src/Io/Io_LTC6813.c

```diff
@@ -0,0 +1,374 @@
+                0x762C, 0xFC41, 0x39D8, 0x32EA, 0xF773, 0x248E, 0xE117, 0xEA25, 0x2FBC,
+                0x846,  0xCDDF, 0xC6ED, 0x374,  0xD089, 0x1510, 0x1E22, 0xDBBB, 0xAF8,
+                0xCF61, 0xC453, 0x1CA,  0xD237, 0x17AE, 0x1C9C, 0xD905, 0xFEFF, 0x3B66,
+                0x3054, 0xF5CD, 0x2630, 0xE3A9, 0xE89B, 0x2D02, 0xA76F, 0x62F6, 0x69C4,
+                0xAC5D, 0x7fA0, 0xBA39, 0xB10B, 0x7492, 0x5368, 0x96F1, 0x9DC3, 0x585A,
+                0x8BA7, 0x4E3E, 0x450C, 0x8095}
+};
+// clang-format on
+
+#define NUM_OF_CELLS_PER_LTC6813_REGISTER_GROUP 3U
+#define NUM_OF_CMD_BYTES 4U
+#define NUM_OF_PEC15_BYTES_PER_CMD 2U
+#define NUM_OF_CELL_VOLTAGE_RX_BYTES 8U
+
+#define ADCOPT 2U
+#define DCP 0U
+#define CELL_CH_ALL 0U
+#define REFON 1U
+#define DTEN 0U
+#define CELL_UNDERVOLTAGE_THRESHOLD 0x4E1
+#define CELL_OVERVOLTAGE_THRESHOLD 0x8CA
+
+#define WRCFGA 0x01
+#define ADCV (0x260 + (ADCOPT << 7) + (DCP << 4) + CELL_CH_ALL)
+#define PLADC 0x1407
+
+struct LTC6813
+{
+    // A pointer to the SPI interface
+    struct SharedSpi *spi;
+
+    // A counter used to count the number of times a PEC error occurs.
+    uint8_t pec15_error_counter;
+
+    // An array containing cell voltages for each LTC6813 IC.
+    uint16_t cell_voltages[NUM_OF_LTC6813][NUM_OF_CELLS_PER_LTC6813];
+};
+
+static struct LTC6813 ltc_6813;
+
+/**
+ * Calculate the 15-bit packet error code (PEC15) for the given data buffer.
+ * @param data A pointer to the buffer containing data used to calculate the
+ * PEC15 code.
+ * @param size The number of data elements used to calculate the PEC15 code
+ * (Note: this number can be a positive integer less than or equal to the size
+ * of the data buffer).
+ * @return The calculated PEC15 code
+ */
+static uint16_t Io_CalculatePec15(uint8_t *data, uint32_t size);
+
+/**
+ * Transition all LTC6813 chips on the daisy chain from the IDLE state to the
+ * READY state.
+ * @return EXIT_CODE_OK if the SCK and NSS pin can be toggled without timing out
+ * or errors. Else, EXIT_CODE_UNIMPLEMENTED.
+ */
+static ExitCode Io_LTC6813_EnterReadyState(void);
+
+/**
+ * Start ADC conversions for all LTC6813 chips on the daisy chain.
+ * @return EXIT_CODE_OK if the command used to toggle the ADC can be sent to the
+ * device without timing out or errors. Else, EXIT_CODE_UNIMPLEMENTED.
+ */
+static ExitCode Io_LTC6813_StartADCConversion(void);
+
+/**
+ * Check if all LTC6813 chips in the daisy chain have all completed converting
+ * analogue cell voltages to digital voltages.
+ * @return EXIT_CODE_OK if all LTC6813 chips on the daisy chain have completed
+ * ADC conversions. EXIT_CODE_TIMEOUT if ADC conversions could not be completed
+ * before timing out. EXIT_CODE_UNIMPLEMENTED if the command sent and received
+ * to check the status of ADC conversions was not transmitted or received
+ * successfully.
+ */
+static ExitCode Io_LTC6813_PollAdcConversion(void);
```

Comment:
Why are all these functions `static`? I don't think it should matter?

Reply:
I made them static just because the user isn't expected to be calling these functions in the app layer. `Io_LTC6813_ReadAllCellRegisterGroups` is the only function that would use the static functions (at least for now).